src/vm/gcinfodecoder.cpp

```diff
@@ -1539,8 +1539,7 @@ void GcInfoDecoder::ReportRegisterToGC(  // AMD64
     LOG((LF_GCROOTS, LL_INFO1000, "Reporting " FMT_REG, regNum ));
 
     OBJECTREF* pObjRef = GetRegisterSlot( regNum, pRD );
-
-#ifdef FEATURE_PAL
+#if defined(FEATURE_PAL) && !defined(SOS_TARGET_AMD64) 
```

Comment:
Zhicheng, you could also add !defined(SOS_TARGET_X86) && !defined(SOS_TARGET_ARM) to the #if.

Or add SOS_TARGET to the CMakeFile.txts and use that.